### ISSUE TITLE
False positive: Sorting hint for namespaced scandir

### ISSUE THREAD

**Issue by kelunik** [kalessil/phpinspectionsea#722]
A namespaced `scandir` also triggers the "Disable sorting" warning, even if that variant doesn't have such a parameter. It could also be an entirely different function in theory.

```php
use Amp\File;

/* ... */
    public function delete($name) {
        return call(function () use ($name) {
            foreach (yield File\scandir($this->root . '/' . $name) as $file) {
                yield File\unlink($this->root . '/' . $name . '/' . $file);
            }

            yield File\rmdir($this->root . '/' . $name);
        });
    }
/* ... */
```

**Comment by kelunik** [comment]
The same applies to other checks, such as `mkdir`: https://github.com/kalessil/phpinspectionsea/blob/master/docs/probable-bugs.md#mkdir-race-condition

```php
if (!yield File\isdir($path) && !yield File\mkdir($path, 0644, true) && !yield File\isdir($path)) {
    throw new FilesystemException("Couldn't create certificate directory: '{$path}'");
}
```

**Comment by kalessil** [comment]
Got it, will fix in the next release.

**Comment by kalessil** [comment]
Does race conditions detection works at all with yeld-ing?

**Comment by kelunik** [comment]
What do you mean? In this case it should just be silent, because those are different functions, not because of the yield. A separate check could be added to do a similar thing for Amp's file library, but that's a different issue.

**Comment by kalessil** [comment]
I'm just surprised that this code gets reported:
```php
if (!yield File\isdir($path) && !yield File\mkdir($path, 0644, true) && !yield File\isdir($path)) {
    throw new FilesystemException("Couldn't create certificate directory: '{$path}'");
}
```

**Comment by kalessil** [comment]
Fixed

**Comment by kalessil** [comment]
@kelunik : I just pushed a fresh build into master, can you give it a try, please?

**Comment by kelunik** [comment]
Will do on Monday or so.
